### DISCUSSION TITLE
Conda installation order / known good "recipe"?

### DISCUSSION THREAD

**Discussion by nbidler** [deepmodeling/deepmd-kit#2740] [Q&A]
I have tried a number of times to install deepmd-kit on a Conda virtual environment. I also need to install cudatoolkit and cudnn. 

However, each time I have tried this (even with multiple different version numbers) I keep getting "unsatisfiable conflict" errors.

Yes, even when following the `conda create -n deepmd deepmd-kit=*=*gpu libdeepmd=*=*gpu lammps cudatoolkit=11.6 horovod -c https://conda.deepmodeling.com -c defaults` guide, I have had these errors.

Does anyone have either 

1. A list of packages in a working environment (must use CUDA)
2. A (shorter) list of what packages they installed, and in what order

Either would be appreciated.

**Comment by njzjz** [comment]
Please see #1362 and #1883.

**Reply by nbidler** [reply]
follow-up: tried using solution 
`conda create -n deepmd_tst deepmd-kit=*=*gpu libdeepmd=*=*gpu lammps-dp cudatoolkit=10.1 horovod -c https://conda.deepmodeling.org --dry-run`
Success, installed without conflicts!

But oh no, a new error.

> Traceback (most recent call last):
  File "/share/apps/ana/envs/deepmd_tst/lib/python3.9/site-packages/numpy/__init__.py", line 328, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'typeDict'
/share/apps/ana/envs/deepmd_tst/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:511: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.
  np.object,
Traceback (most recent call last):
BIG LIST OF TRACE HERE
AttributeError: module 'numpy' has no attribute 'object'.
`np.object` was a deprecated alias for the builtin `object`. To avoid this error in existing code, use `object` by itself. Doing this will not modify any behavior and is safe.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

Edit: there is more than one of this same error, I only put one here to save space.

I am going to attempt downgrading NumPy to 1.19 ( just with the command `conda install numpy=1.19` to see if it accepts that first), will update with results

**Reply by nbidler** [reply]
The answer is "It creates a lot of conflicts" (see attached image) so I am going to try other solutions.
Might try `conda create -n deepmd_tst2 deepmd-kit=*=*gpu libdeepmd=*=*gpu lammps-dp cudatoolkit=* horovod numpy=1.19  -c https://conda.deepmodeling.org` to see if the additional restriction works

![image](https://github.com/deepmodeling/deepmd-kit/assets/9604309/b6248c37-f71f-4801-ad57-ba5f4eec5cd6)

**Reply by nbidler** [reply]
current conflicts are:

```
  - feature:/linux-64::__glibc==2.17=0
  - cudatoolkit -> __glibc[version='>=2.17,<3.0.a0']
  - cudatoolkit -> libgcc-ng[version='>=11.2.0'] -> __glibc[version='>=2.17']
  - deepmd-kit[build=*gpu] -> __glibc[version='>=2.17']
  - deepmd-kit[build=*gpu] -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - horovod -> __cuda
  - horovod -> __glibc[version='>=2.17']
  - horovod -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - horovod -> nccl[version='>=2.14.3.1,<3.0a0'] -> __cuda[version='10.2|10.2.*|11.0|11.0.*|>=11.2,<12|11.1|11.1.*']
  - libdeepmd[build=*gpu] -> __glibc[version='>=2.17']
  - libdeepmd[build=*gpu] -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - numpy=1.19 -> libgcc-ng[version='>=10.3.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.17
```

I will try to determine which earlier versions lack this conflict.
